realm/realm-library/src/androidTest/java/io/realm/TypeBasedNotificationsTest.java

```diff
@@ -1807,4 +1809,69 @@ public void run() {
         TestHelper.awaitOrFail(listenerWasCalledOnRealmObject);
         TestHelper.awaitOrFail(listenerWasCalledOnRealmResults);
     }
+
+    // Test modifying realmObjects in RealmObject's change listener
+    public void test_change_realm_objects_map_in_listener() throws InterruptedException {
+        handler.post(new Runnable() {
+            @Override
+            public void run() {
+                realm = Realm.getInstance(configuration);
+
+                realm.beginTransaction();
+                final Owner owner = realm.createObject(Owner.class);
+                owner.setCat(realm.createObject(Cat.class));
+                realm.commitTransaction();
+
+                owner.addChangeListener(new RealmChangeListener() {
+                    @Override
+                    public void onChange() {
+                        Cat cat = owner.getCat();
+                        // Keep calling addToRealmObjects just in case it won't be called in getCat in the future
```

Comment:
This seems a bit dangerous. Isn't it better that this unit test start failing if addToRealmObjects no longer gets called?